File: Python/guessing.py
# The `import random` statement is importing the `random` module in Python. This module provides
# functions for generating random numbers, selecting random elements from a sequence, and other
# random-related operations.
import random
"""
guesses = input("How many guesses do you want to make? ")
guess_number = random.randint(1, 100)

for i in range(int(guesses)):
    guess = int(input("Guess a number between 1 and 100: "))
    if guess == guess_number:
        print("You got it right good work!")
        retry = input("Do you want to play again? (yes/no): ")
        if retry.lower() != "yes":
            break
    elif guess < guess_number:
        print("Your guess is too low try again!")
    elif guess > guess_number:
        print("Your guess is too high try again!")
    else:
        print("Your guess is not a number try again!")
 """
# The code is implementing a simple rock-paper-scissors game.

options = ("rock", "paper", "scissors")
player= None
computer = random.choice(options)

while player not in options:
    player = input("Enter rock, paper, or scissors: ")

if player == computer:
    print("It's a tie!")
elif player == "rock" and computer == "scissors":
    print("You won!")
elif player == "paper" and computer == "rock":
    print("You won!")
elif player == "scissors" and computer == "paper":
    print("You won!")
else:
    print("You lost!")

File: c++/2023/guessing/guessing.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>

int main()
{
  srand(time(0));
  int secretNumber = rand() % 100 + 1;

  int guess;
  int attempts = 0;

  std::cout << "Welcome to the Guessing Game!\n";
  std::cout << "I'm thinking of a number between 1 and 100.\n";

  do
  {
    std::cout << "Enter your guess: ";
    std::cin >> guess;

    attempts++;

    if (guess < secretNumber)
    {
      std::cout << "Too low! Try again.\n";
    }
    else if (guess > secretNumber)
    {
      std::cout << "Too high! Try again.\n";
    }
    else
    {
      std::cout << "Congratulations! You guessed the number in " << attempts << " attempts.\n";
    }
  } while (guess != secretNumber);

  return 0;
}